src/main/java/gov/nasa/jpl/mbee/mdk/systems_reasoner/actions/ImportCSVAction.java

```diff
@@ -0,0 +1,416 @@
+    private static Namespace container;
+
+    public ImportCSVAction(Classifier classifier) {
+        super(DEFAULT_ID);
+        this.selectedClassifier = classifier;
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent arg0) {
+        final List<java.lang.Class<?>> types = new ArrayList<>();
+        types.add(com.nomagic.uml2.ext.magicdraw.classes.mdkernel.Class.class);
+        types.add(com.nomagic.uml2.ext.magicdraw.classes.mdkernel.Package.class);
+        types.add(Model.class);
+
+
+        GUILog gl = Application.getInstance().getGUILog();
+        String separator = getSeparator();
+        if (separator == null) {
+            return;
+        }
+        row = 0;
+        JFileChooser choose = new JFileChooser();
+        choose.setDialogTitle("Open CSV file");
+        choose.setFileSelectionMode(JFileChooser.FILES_ONLY);
+        HashSet<Classifier> createdElements = new HashSet<>();
+        int retval = choose.showOpenDialog(null);
+        if (retval == JFileChooser.APPROVE_OPTION) {
+            if (choose.getSelectedFile() != null) {
+                File savefile = choose.getSelectedFile();
+                try {
+
+                    final Frame dialogParent = MDDialogParentProvider.getProvider().getDialogParent();
+                    final ElementSelectionDlg dlg = ElementSelectionDlgFactory.create(dialogParent);
+                    dlg.setTitle("Select container for generated elements:");
+                    final SelectElementTypes set = new SelectElementTypes(null, types, null, null);
+                    final SelectElementInfo sei = new SelectElementInfo(true, false, Application.getInstance().getProject().getModel().getOwner(), true);
+                    ElementSelectionDlgFactory.initSingle(dlg, set, sei, selectedClassifier.getOwner());
+                    dlg.setSelectionMode(SelectionMode.SINGLE_MODE);
+                    dlg.setVisible(true);
+                    if (dlg.isOkClicked() && dlg.getSelectedElement() != null && dlg.getSelectedElement() instanceof Namespace) {
+                        container = (Namespace) dlg.getSelectedElement();
+                        SessionManager.getInstance().createSession("change");
+                        gl.log("Starting CSV import.");
+                        CSVReader reader = new CSVReader(new FileReader(savefile), separator.charAt(0));
+                        importFromCsv(reader, createdElements);
+                        checkForRedefinedElements(createdElements);
+                        reader.close();
+                        SessionManager.getInstance().closeSession();
+                        checkForAssociationInheritance(createdElements);
+                        gl.log("CSV import finished.");
+                    }
+                } catch (IOException ex) {
+                    gl.log("CSV import failed:");
+                    SessionManager.getInstance().cancelSession();
+                    gl.log(ex.getMessage());
```

Comment:
Exceptions should be printed to console/log with e.printStackTrace(), not to the notification window.